pegasus/helpers/hourofcode_helpers.rb

```diff
@@ -93,6 +93,33 @@ def hoc_detect_language
   nil
 end
 
+def load_supported_locales
+  Dir.glob(pegasus_dir('cache', 'i18n', '*.yml')).map do |i|
+    File.basename(i, '.yml').downcase
+  end.sort
+end
+HOC_LOCALES = load_supported_locales
+
+def language_to_locale(language)
+  case language
+  when 'en'
+    return 'en-US'
+  when 'es'
+    return 'es-ES'
+  when 'fa'
+    return 'fa-IR'
+  else
+    language = language.to_s.downcase
+    return nil unless locale = HOC_LOCALES.find{|i| i == language || i.split('-').first == language}
+    parts = locale.split('-')
+    return "#{parts[0].downcase}-#{parts[1].upcase}"
+  end
+end
+
+def hoc_locale
+  language_to_locale(@language)
+end
+
```

Comment:
@tanyaparker Will this generate the correct locale list that correlates to what's going to appear in apps/i18n/tutorialExplorer?